### PR TITLE
Parallel jobs: get cost data from dynamic nodes

### PR DESCRIPTION
**Pull request by MightyNerdEric**
Previously, the LF's job cost scripts could not take into account the
agents that are dynamically allocated when an "agent" block is used
within a parallel stage. A new function has been added to the LF's
pipeline library that is intended to be run in a post block after any
parallel stage that uses a new agent.

This change also includes some minor whitespace cleanup in affected
files, as well as adding a license to edgeXGeneric and updating
copyright dates.

Issue: LF-Jira RELENG-3207
Signed-off-by: Eric Ball <eball@linuxfoundation.org>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Sandbox Testing
Test Links :

### CODE REVIEW

**Comment by MightyNerdEric** [comment]
A test of this change can be seen here: https://jenkins.edgexfoundry.org/sandbox/view/All/job/sample-service-test/job/master/20/
This job is running edgeXBuildGoApp, with the additions that are suggested in this change.

The end result is not terribly obvious, but if you look at the cost.csv file that is generated, you can see in the last column that there is a stack-cost. If you look at any jobs currently running in prod, you'll see that that value is always 0, even though on edgeXBuildGoApp runs, two extra agents are being spun up.

File: vars/edgeXBuildGoParallel.groovy [hunk 1]
```diff
@@ -467,6 +474,7 @@ def toEnvironment(config) {
 
     // def _snapChannel           = config.snapChannel ?: 'latest/edge'
     def _buildSnap             = edgex.defaultFalse(config.buildSnap)
+    def _snykDockerScan        = edgex.defaultFalse(config.snykDockerScan)
```
Comment (soda480) [inline vars/edgeXBuildGoParallel.groovy:477]: why was this line added `def _snykDockerScan        = edgex.defaultFalse(config.snykDockerScan)`?

**Review by soda480 — COMMENTED** [review]
LGTM - just one question regarding the addition of the snykScan variable.

File: vars/edgeXBuildGoParallel.groovy (hunk 1 above)
Comment (MightyNerdEric) [reply vars/edgeXBuildGoParallel.groovy]: Whoops, I had to do merge conflict resolution, and just screwed that one up. Great catch, thanks! I've pushed a fixed version.

**Review by soda480 — APPROVED** [review]
LGTM

**Review by ernestojeda — APPROVED** [review]
LGTM.
